File: Algorithms/01-Warmup/05-plus-minus/05.rb
#!/usr/bin/env ruby

def plus_minus(arr)
    pos = 0
    neg = 0

    arr.each do |i|
        if i > 0
            pos += 1
        elsif i < 0
            neg += 1
        end
    end
    puts pos / arr.length.to_f
    puts neg / arr.length.to_f
    puts (arr.length - (pos + neg)) / arr.length.to_f
end

n = gets.strip.to_i
arr = gets.strip
arr = arr.split(" ").map(&:to_i)
plus_minus(arr)

File: Algorithms/01-Warmup/05-plus-minus/05.py
#!/usr/bin/env python


def plusMinus(arr):
    pos = 0
    neg = 0

    for i in arr:
        if i > 0:
            pos += 1
        elif i < 0:
            neg += 1
    print(pos / len(arr))
    print(neg / len(arr))
    print((len(arr) - (pos + neg)) / len(arr))


if __name__ == "__main__":
    n = int(input().strip())
    arr = list(map(int, input().strip().split(' ')))
    plusMinus(arr)
